applications/DG-Max/DGMaxProgramUtils.cpp

```diff
@@ -148,4 +149,76 @@ PointPath<DIM> parsePath(const std::string& path) {
 template PointPath<1> parsePath(const std::string& path);
 template PointPath<2> parsePath(const std::string& path);
 template PointPath<3> parsePath(const std::string& path);
+
+std::unique_ptr<ZoneInfoStructureDefinition> readZonedDescription(
+    std::istream& file, char separator) {
+    std::string line;
+    std::vector<std::regex> zoneRegexes;
+    std::vector<double> zoneEpsilons;
+    std::size_t lineNumber = 0;
+    while (!file.eof()) {
+        std::getline(file, line, separator);
+        std::size_t commaLoc = line.find_last_of(',');
+        logger.assert_always(commaLoc != std::string::npos,
+                             "No comma found on line %: line", lineNumber,
+                             line);
+        std::string regexStr = line.substr(0, commaLoc);
+        std::regex regex(regexStr);
+
+        std::string epsilonStr = line.substr(commaLoc + 1);
+        logger.assert_always(!epsilonStr.empty(),
+                             "No value after the comma on line %: %",
+                             lineNumber, line);
+        std::size_t idx;
+        double epsilon = 0.0;
+        try {
+            epsilon = std::stod(epsilonStr, &idx);
+        } catch (std::invalid_argument&) {
+            logger.assert_always(false, "Invalid epsilon value on line %: %",
+                                 lineNumber, line);
+        }
+        logger.assert_always(idx == epsilonStr.size(),
+                             "Trailing data after epsilon on line %: %",
+                             lineNumber, line);
+        zoneRegexes.push_back(regex);
+        zoneEpsilons.push_back(epsilon);
+
+        lineNumber++;
+    }
+    return std::make_unique<ZoneInfoStructureDefinition>(zoneRegexes,
+                                                         zoneEpsilons);
+}
+
+std::unique_ptr<StructureDescription> determineStructureDescription(
+    const std::string& input, std::size_t dim) {
+
+    // Test to see if the input is a number, by trying to parse it
+    char* end;
+    long value = std::strtol(input.c_str(), &end, 10);
+    if (!*end) {
```

Comment:
Can we make this a small function "convertToLong" or so

Reply:
What should it do beyond what `std::strtol` is already doing?

Note to self: I should reread the docs on it and fix the end pointer.

Reply:
Yes, but it is not really nice to read because of all the pointers and basically you want to convert a string to a long